crates/remote/src/ssh_session.rs

```diff
@@ -1999,58 +2000,31 @@ impl SshRemoteConnection {
             Ok(())
         }
 
-        if platform.arch == std::env::consts::ARCH && platform.os == std::env::consts::OS {
-            delegate.set_status(Some("Building remote server binary from source"), cx);
-            log::info!("building remote server binary from source");
-            run_cmd(Command::new("cargo").args([
-                "build",
-                "--package",
-                "remote_server",
-                "--features",
-                "debug-embed",
-                "--target-dir",
-                "target/remote_server",
-            ]))
-            .await?;
-
-            delegate.set_status(Some("Compressing binary"), cx);
-
-            run_cmd(Command::new("gzip").args([
-                "-9",
-                "-f",
-                "target/remote_server/debug/remote_server",
-            ]))
-            .await?;
-
-            let path = std::env::current_dir()?.join("target/remote_server/debug/remote_server.gz");
-            return Ok(path);
-        }
         let Some(triple) = platform.triple() else {
             anyhow::bail!("can't cross compile for: {:?}", platform);
         };
-        smol::fs::create_dir_all("target/remote_server").await?;
-
-        if build_remote_server.contains("cross") {
-            delegate.set_status(Some("Installing cross.rs for cross-compilation"), cx);
-            log::info!("installing cross");
-            run_cmd(Command::new("cargo").args([
-                "install",
-                "cross",
-                "--git",
-                "https://github.com/cross-rs/cross",
-            ]))
-            .await?;
+        let mut rust_flags = match std::env::var("RUSTFLAGS") {
+            Ok(val) => val,
+            Err(VarError::NotPresent) => String::new(),
+            Err(e) => {
+                log::error!("Failed to get env var `RUSTFLAGS` value: {e}");
+                String::new()
+            }
+        };
+        if platform.os == "linux" {
+            rust_flags.push_str(" -C target-feature=+crt-static");
+        }
+        if build_remote_server.contains("mold") {
+            rust_flags.push_str(" -C link-arg=-fuse-ld=mold");
+        }
 
-            delegate.set_status(
-                Some(&format!(
-                    "Building remote server binary from source for {} with Docker",
-                    &triple
```

Comment:
It's not bad to mention that `triple` is it? 
Any reason to omit?

Reply:
I'm not sure to understand your point.
I don't have remove this, but I change the scope (so indentation changed) and the diff is a little lost.

Why do you think about split the build_local function in several smaller functions before ? This can help diff reading.

Reply:
Just, can we put that `tripe` back in the status output?
It's removed in the PR for some reason.

Reply:
It's not removed, the leading space change, so the diff mark it as added + removed. You can verify with a search.

Reply:
Are you interested in me splitting it into smaller commits to make the change more readable?